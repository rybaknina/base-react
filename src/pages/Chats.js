import ControlPanel from "../components/ControlPanel";
import ChatList from "../components/ChatList";
import Paper from "@mui/material/Paper";
import Grid from "@mui/material/Grid";
import Typography from "@mui/material/Typography";
import { Button, useTheme } from "@mui/material";
import AddBoxIcon from "@mui/icons-material/AddBox";

const Chats = ({ chats, addMessage }) => {
	const theme = useTheme();
	return (
		<>
			<Grid container>
				<Grid item xs={12}>
					<Typography
						variant="h6"
						style={{
							backgroundColor: theme.palette.primary.light,
							color: theme.palette.primary.contrastText,
						}}
					>
						Список чатов
						<Button
							style={{ color: theme.palette.primary.contrastText }}
							// todo onClick={addChat}
						>
							<AddBoxIcon />
						</Button>
					</Typography>
				</Grid>
			</Grid>
			<Grid container component={Paper} style={{ width: "100%", height: "74vh" }}>
				<ChatList chats={chats} />
			</Grid>
			<ControlPanel addMessage={addMessage} />
		</>
	);
};

export default Chats;
