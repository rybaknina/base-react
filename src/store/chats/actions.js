export const ADD_CHAT = "CHATS::ADD_CHAT";
export const DEL_CHAT = "CHATS::DEL_CHAT";

export const addChat = (name) => ({
	type: ADD_CHAT,
	payload: name,
});

export const deleteChat = (chatId) => ({
	type: DEL_CHAT,
	payload: chatId,
});
