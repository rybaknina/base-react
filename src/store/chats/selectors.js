const getChatList = (state) => {
	return state.chats.chatList;
};

export default getChatList;
