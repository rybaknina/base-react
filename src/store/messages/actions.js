import { AUTHOR, TIME_IN_SECONDS } from "../../constants/common";
import _uniqueId from "lodash/uniqueId";

export const ADD_MESSAGE = "MESSAGES::ADD_MESSAGE";
export const ADD_MESSAGE_WITH_SAGA = "MESSAGES::ADD_MESSAGE_WITH_SAGA";

export const addMessage = (chatId, message) => ({
	type: ADD_MESSAGE,
	payload: { chatId, message },
});

export const addMessageWithSaga = (chatId, message) => ({
	type: ADD_MESSAGE_WITH_SAGA,
	payload: { chatId, message },
});

export const addMessageWithThunk = (chatId, message) => (dispatch) => {
	dispatch(addMessage(chatId, message));

	if (message.author !== AUTHOR.bot) {
		const botMessage = {
			id: _uniqueId("bot_"),
			text: "Hello, friend! I'm from thunk",
			author: AUTHOR.bot,
		};
		setTimeout(() => {
			dispatch(addMessage(chatId, botMessage));
		}, TIME_IN_SECONDS);
	}
};
