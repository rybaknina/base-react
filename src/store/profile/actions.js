export const TOGGLE_NAME = "TOGGLE_NAME";
export const CHECK_ME = "CHECK_ME";
export const UPDATE_NAME = "UPDATE_NAME";

export const changeVisible = {
	type: TOGGLE_NAME,
};

export const changeCheck = {
	type: CHECK_ME,
};

export const updateName = (name) => ({
	type: UPDATE_NAME,
	payload: name,
});
