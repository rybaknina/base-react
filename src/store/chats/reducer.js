import { ADD_CHAT, DEL_CHAT } from "./actions";
import _uniqueId from "lodash/uniqueId";

const initialState = {
	chatList: [],
};

const chatsReducer = (state = initialState, action) => {
	switch (action.type) {
		case ADD_CHAT: {
			const newChat = {
				id: _uniqueId("chat_"),
				name: action.payload,
			};
			return {
				...state,
				chatList: [...state.chatList, newChat],
			};
		}
		case DEL_CHAT: {
			return {
				...state,
				chatList: [...state.chatList.filter((chat) => chat.id !== action.payload)],
			};
		}
		default:
			return state;
	}
};

export default chatsReducer;
