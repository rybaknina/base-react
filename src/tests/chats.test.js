import { ADD_CHAT, DEL_CHAT, UPDATE_CHATS } from "../store/chats/actions";
import chatsReducer from "../store/chats/reducer";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Chats from "../pages/Chats";

describe("chats tests", () => {
	describe("chat reducer test", () => {
		it("return state with chatList after addChat action", () => {
			const expected = {
				chatList: [
					{
						id: "chat_1",
						name: "name",
					},
				],
			};
			const received = chatsReducer({ chatList: [] }, { type: ADD_CHAT, payload: "name" });
			expect(received).toEqual(expected);
		});

		it("return state with empty chatList after delChat action", () => {
			const expected = {
				chatList: [],
			};
			const received = chatsReducer(
				{ chatList: [{ id: "chat_1", name: "name" }] },
				{ type: DEL_CHAT, payload: "chat_1" },
			);
			expect(received).toEqual(expected);
		});

		it("return state with updated chatList after updateChat action", () => {
			const chats = [
				{
					id: "chat_1",
					name: "name_1",
				},
				{
					id: "chat_2",
					name: "name_2",
				},
			];
			const expected = {
				chatList: chats,
			};
			const received = chatsReducer({ chatList: [] }, { type: UPDATE_CHATS, chats });
			expect(received).toEqual(expected);
		});
	});

	describe("snapshot chats component", () => {
		const initialState = {
			chats: { chatList: [{ id: "chat_1", name: "chat_1" }] },
			messages: { messageList: [{ id: "123", name: "name" }] },
			profile: {
				name: "testName",
			},
		};
		const mockStore = configureStore();
		it("snapshot test", () => {
			let store = mockStore(initialState);
			const component = render(
				<Provider store={store}>
					<Chats />
				</Provider>,
				{ wrapper: MemoryRouter },
			);
			expect(component).toMatchSnapshot();
		});
	});
});
