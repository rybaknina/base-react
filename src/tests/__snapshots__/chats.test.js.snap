// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chats tests snapshot chats component snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            style="background-color: rgb(66, 165, 245); color: rgb(255, 255, 255); padding: 5px 15px;"
          >
            Список чатов
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiGrid-root MuiGrid-container css-j1q43y-MuiPaper-root-MuiGrid-root"
        style="width: 100%; height: 76vh;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          style="border-right: 1px solid #e0e0e0; text-align: center;"
        >
          <button
            aria-label="add chat"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1ylii70-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              style="padding: 5px;"
            >
              Add chat
            </h6>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            style="height: 68vh; overflow: auto;"
          >
            <a
              href="/chats/chat_1"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rki8sg-MuiListItem-root"
              >
                <div
                  class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="FolderIcon"
                      focusable="false"
                      style="color: rgb(123, 31, 162);"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                  style="word-wrap: break-word;"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    chat_1
                  </span>
                </div>
                <div
                  class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
                >
                  <button
                    aria-label="delete"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="DeleteIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </li>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
              />
            </a>
          </ul>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-1qpzhgb-MuiGrid-root"
          style="padding-top: 6px;"
        >
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            style="height: 73vh; overflow: auto; background-color: rgb(66, 165, 245);"
          />
        </div>
      </div>
      <div
        style="background-color: rgb(66, 165, 245);"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          style="padding: 20px 15px;"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 css-ykovpn-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              style="margin-right: 10px; color: rgb(21, 101, 192);"
            >
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-2"
                  placeholder="текст сообщения"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            align="right"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
          >
            <button
              aria-label="add"
              class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary css-1suf56w-MuiButtonBase-root-MuiFab-root"
              style="background-color: rgb(21, 101, 192);"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SendIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          style="background-color: rgb(66, 165, 245); color: rgb(255, 255, 255); padding: 5px 15px;"
        >
          Список чатов
        </h6>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiGrid-root MuiGrid-container css-j1q43y-MuiPaper-root-MuiGrid-root"
      style="width: 100%; height: 76vh;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        style="border-right: 1px solid #e0e0e0; text-align: center;"
      >
        <button
          aria-label="add chat"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1ylii70-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            style="padding: 5px;"
          >
            Add chat
          </h6>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          style="height: 68vh; overflow: auto;"
        >
          <a
            href="/chats/chat_1"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rki8sg-MuiListItem-root"
            >
              <div
                class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FolderIcon"
                    focusable="false"
                    style="color: rgb(123, 31, 162);"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                style="word-wrap: break-word;"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  chat_1
                </span>
              </div>
              <div
                class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
              >
                <button
                  aria-label="delete"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DeleteIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
          </a>
        </ul>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-1qpzhgb-MuiGrid-root"
        style="padding-top: 6px;"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          style="height: 73vh; overflow: auto; background-color: rgb(66, 165, 245);"
        />
      </div>
    </div>
    <div
      style="background-color: rgb(66, 165, 245);"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        style="padding: 20px 15px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 css-ykovpn-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            style="margin-right: 10px; color: rgb(21, 101, 192);"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-2"
                placeholder="текст сообщения"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          align="right"
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
        >
          <button
            aria-label="add"
            class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary css-1suf56w-MuiButtonBase-root-MuiFab-root"
            style="background-color: rgb(21, 101, 192);"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SendIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
